### DISCUSSION TITLE
How to get info about player?

### DISCUSSION THREAD

**Discussion by ScarletRedMan** [geode-sdk/sdk#88] [Help] [answered]
I wanna get information about player inside any scene in the game. If it is possible, then I want to get such player parameters as: stars, name, demons, hashed password (not for stealing account), orbs, diamonds, icon kit, etc.

**Comment by camila314** [comment, marked as answer]
GameStatsManager class has the stats, GJAccountManager has the account details. Password isn't hashed lol

**Reply by ScarletRedMan** [reply]
Thanks

Confused the word 'hashed' and 'encrypted' :D

**Reply by camila314** [reply]
oh you're not gonna believe this, it's not encrypted either

**Reply by ScarletRedMan** [reply]
I do not know in what form it is stored inside the game, but when I wrote a server on the Spring Framework, the password transfer there was encrypted with a primitive xor cipher

**Reply by camila314** [reply]
robtop stores it in plaintext 💀

**Reply by ScarletRedMan** [reply]
I looked at the classes that you wrote to me and I didn't understand how to interact with them. How do I get one of the classes?

**Reply by HJfod** [reply]
**Manager classes** (including the ones listed) are defined by only **a single shared instance of that class existing at a time**. All managers should have a static getter method in them, which is used to get that shared global instance. Usually in Geode our principle is that the function is called `get`, however RobTop usually calls them `sharedState` (or something similar) so see if the classes have a `get` method, and if not, see if they have something like `sharedState` or similar.

Example:
```cpp
// Get the name of the currently logged in user
auto username = GJAccountManager::sharedState()->m_username;
// Get the password of the currently logged in user
auto password = GJAccountManager::sharedState()->m_password;
```

**Reply by ScarletRedMan** [reply]
Please tell me how to get the number of player stars? I can't figure out how to get player parameters using GameStateManager.
